frontend/fingertips-frontend/components/organisms/Inequalities/index.tsx

```diff
@@ -1,17 +1,142 @@
-import { InequalitiesTable } from '@/components/molecules/Inequalities/Table';
-import { HealthDataForArea } from '@/generated-sources/ft-api-client';
+import { InequalitiesBarChartTable } from '@/components/molecules/Inequalities/BarChart/Table';
+import { InequalitiesTable } from '@/components/molecules/Inequalities/LineChart/Table';
+import {
+  HealthDataForArea,
+  HealthDataPoint,
+} from '@/generated-sources/ft-api-client';
 import React from 'react';
+import {
+  getYearDataGroupedByInequalities,
+  groupHealthDataByYear,
+  mapToKey,
+} from './inequalitiesHelpers';
+import { H4 } from 'govuk-react';
+import { TabContainer } from '@/components/layouts/tabContainer';
 
 interface InequalitiesProps {
   healthIndicatorData: HealthDataForArea;
 }
 
+export interface InequalitiesLineChartTableData {
+  areaName: string;
+  rowData: InequalitiesTableRowData[];
+}
+
+export interface InequalitiesBarChartTableData {
+  areaName: string;
+  data: InequalitiesTableRowData;
+}
+
+export interface RowDataFields {
+  count?: number;
+  value?: number;
+  lower?: number;
+  upper?: number;
+}
+
+export interface InequalitiesTableRowData {
+  [key: string]: number | RowDataFields | undefined;
```

Comment:
Can this just be:
```suggestion
  [key: string]: RowDataFields | undefined;
```

I'm not sure you're ever using the form where the value is a number, are you?

Reply:
I think this is some typing strangeness. If you make this interface as I suggested above it should mean you don't need to cast the values to `RowDataFields` in the table components. It just needs a bit of reworking of the mapping function. I had a quick plan and think I have something that works, so let me know if you'd like some help with it.

Reply:
the period key is a number, while `Persons`, `Male`, and `Female` are of type `RowDataFields`. So deleting the number type would cause the object mapping to break. Happy speak about your plan to change the mapping function

Reply:
Could `period` be defined as part of `RowDataFields`?

Reply:
I'd rather keep `period` apart from the `RowDataFields` because each period corresponds to a different row under the period column, while the `RowDataFields` defines the data that the inequalities hold. Having `period` as part of `RowDataFields` duplicates it across each inequality for each row and would require extra logic to extract its value. I've taken a different approach that's closer to what was there initially, let me know if you're not happy about it.

Reply:
I'm happy with the approach you've taken :)